### PR TITLE
[Listener] Filter out events in logic

### PR DESCRIPTION
**Pull request by franckc**
The listener now subscribes to all events emitted by the marketplace contract and filters out events that are not related to dshop in the logic.

Previously the listener was filtering out events by specifying the dshop listing ids in the query it was making to the network. This resulted in the possible issue:
 1. Shop creation flow start from superadmin, a listing is created on the marketplace
 2. Listener picks up the new listing event but the listing is not associated yet with the new shop in the DB (that only happens in step 3 below). So the listener ignores the event.
 3. Shop creation flow finishes, listing id gets associated with shop in the DB
 4. Transactions for the new shop are ignored since the listener is not listening for events on the new shop's listing

Added unit tests to check the new logic.

### CODE REVIEW

File: backend/scripts/backfill.js [hunk 1]
```diff
@@ -181,7 +181,7 @@ async function handleEvents(listingIdFull) {
     order: [['block_number', 'ASC']]
   })
   for (const event of events) {
-    await insertOrderFromEvent({
+    await processDShopEvent({
       offerId: `${listingIdFull}-${event.offerId}`,
```
Comment (nick) [inline backend/scripts/backfill.js:185]: I guess we don't need this any more since it may not be an offer event?

**Comment by nick** [comment]
Could it also be worth adding a test for a non Order event, like CreateListing? Just to make sure that doesn't mess with anything.

File: backend/scripts/backfill.js (hunk 1 above)
Comment (franckc) [reply backend/scripts/backfill.js]: Good point. Removed the offerId argument from this method.

**Comment by franckc** [comment]
Addressed review comments, added a test for non-Offer event ListingUpdated.

**Pull request merged**